ouroboros-consensus-cardano/test/Test/Consensus/Cardano/Generators.hs

```diff
@@ -292,33 +293,89 @@ instance (sc ~ TPraosMockCrypto h, HashAlgorithm h, forall a. Arbitrary (Hash h
       ]
 
 instance Arbitrary (Some QueryAnytime) where
-  arbitrary = return $ Some EraStart
+  arbitrary = return $ Some GetEraStart
+
+instance Arbitrary (Some (QueryHardFork (CardanoEras sc))) where
+  arbitrary = return $ Some GetInterpreter
 
 instance (sc ~ TPraosMockCrypto h, HashAlgorithm h)
       => Arbitrary (WithVersion (HardForkNodeToClientVersion (CardanoEras sc))
                                 (SomeBlock Query (CardanoBlock sc))) where
   arbitrary = frequency
-      [ (9, arbitraryNodeToClient injByron injShelley)
+      [ (1, arbitraryNodeToClient injByron injShelley)
+      , (1, WithVersion
+              <$> (getHardForkEnabledNodeToClientVersion <$> arbitrary)
+              <*> (injAnytimeByron <$> arbitrary))
       , (1, WithVersion
               <$> (getHardForkEnabledNodeToClientVersion <$> arbitrary)
               <*> (injAnytimeShelley <$> arbitrary))
+      , (1, WithVersion
+              <$> (getHardForkEnabledNodeToClientVersion <$> arbitrary)
+              <*> (injHardFork <$> arbitrary))
       ]
     where
       injByron          (SomeBlock query) = SomeBlock (QueryIfCurrentByron   query)
       injShelley        (SomeBlock query) = SomeBlock (QueryIfCurrentShelley query)
+      injAnytimeByron   (Some      query) = SomeBlock (QueryAnytimeByron     query)
       injAnytimeShelley (Some      query) = SomeBlock (QueryAnytimeShelley   query)
+      injHardFork       (Some      query) = SomeBlock (QueryHardFork         query)
+
+instance Arbitrary History.EraEnd where
+  arbitrary = oneof
+      [ History.EraEnd <$> arbitrary
+      , return History.EraUnbounded
+      ]
+
+instance Arbitrary History.SafeBeforeEpoch where
+  arbitrary = oneof
+      [ return History.NoLowerBound
+      , History.LowerBound . EpochNo <$> arbitrary
+      , return History.UnsafeUnbounded
+      ]
+
+instance Arbitrary History.SafeZone where
+  arbitrary = History.SafeZone
+      <$> arbitrary
+      <*> arbitrary
+
+instance Arbitrary History.EraParams where
+  arbitrary = History.EraParams
+      <$> (EpochSize <$> arbitrary)
+      <*> arbitrary
+      <*> arbitrary
+
+instance Arbitrary History.EraSummary where
+  arbitrary = History.EraSummary
+      <$> arbitrary
+      <*> arbitrary
+      <*> arbitrary
+
+instance Arbitrary (History.Interpreter (CardanoEras sc)) where
+  arbitrary = History.mkInterpreter . History.Summary <$> genNonEmpty
+    where
+      genNonEmpty :: Arbitrary a => Gen (NonEmpty (CardanoEras sc) a)
+      genNonEmpty = oneof
+          [ NonEmptyOne  <$> arbitrary
+          , NonEmptyCons <$> arbitrary <*> (NonEmptyOne <$> arbitrary)
+          ]
```

Comment:
Write an `Arbitrary` instance for `NonEmpty` because we'll forget to update this when adding the next era.